core/net/ipv6/uip-nd6.c

```diff
@@ -682,7 +684,6 @@ uip_nd6_ra_output(uip_ipaddr_t * dest)
   nd6_opt_offset = UIP_ND6_RA_LEN;
 
 
-#if !UIP_CONF_ROUTER
```

Comment:
I think it will useless to add this macro. Indeed, at line 562 and 563, there are 2 macros saying that it needs to be a router and it sends RA. The code design is like this:
```
#if UIP_CONF_ROUTER
#if UIP_ND6_SEND_RA
static void
rs_input(void)
{
...
}
...
void
uip_nd6_ra_output(uip_ipaddr_t * dest)
{
...
}
#endif /* UIP_ND6_SEND_RA */
#endif /* UIP_CONF_ROUTER */
```

So the macro ``(!UIP_CONF_ROUTER || UIP_ND6_SEND_RA)`` will be always True because ``UIP_ND6_SEND_RA`` is set to True.

I think both macros are well written at line 562 and 563 because it needs to be a router to send a router advertisement (``UIP_CONF_ROUTER``). And all routers shouldn't send this type of message. We need to declare that this router must send RAs (``UIP_ND6_SEND_RA``).
 
About RA, we need to send Prefix Information Option (PIO) like described in the RFC4861. If you wouldn't like to send PIOs, you can set the flag ``advertise`` to 0 on the ``uip_ds6_prefix_list`` prefixes table.